username_api.py

```diff
@@ -12,11 +14,56 @@
 
 patterns = yaml.load(open('websites.yml'))
 
-def check_username(website, username):
-	url = patterns['urls'].get(website, 'https://{w}.com/{u}').format(
+def get_profile_url(website, username):
+	return patterns['urls'].get(website, 'https://{w}.com/{u}').format(
 		w=website,
-		u=username
-	)
+		u=username)
+
+def get_avatar(website, username):
+	data = patterns['avatar'][website]
+
+	if not data:
+		return None
+
+	url = get_profile_url(website, username)
+	if 'url' in data:
+		url = data['url'].format(u=username)
+
+	response = r.get(url)
+	if response.status_code == 404:
+		return None
+
+	if data == 'opengraph':
+		# Look in metadata for image.
+		soup = BeautifulSoup(response.text, 'html.parser')
+		result = [item.attrs['content'] for item in soup('meta')
+				  if item.has_attr('property') and
+				  item.attrs['property'].lower() == 'og:image']
+		if not result or not result[0]:
+			return None
+		result = result[0]
+	elif 'property' in data:
+		# Searches for "`property`": "`link`"
+		regex = re.compile('[\'\"]' + re.escape(data['property']) +
+						   '[\'\"]:(\s)?[\'\"](?P<link>[^\s]+)[\'\"]')
+		result = re.search(regex, response.text)
+		if not result:
+			return None
+		result = result.group('link')
+	elif (response.headers.get('content-type') and
```

Comment:
I would use `response.headers.get('content-type', '').startswith('image/')` to make it more readable and only perform slightly worse in the very unlikely scenario that one of these sites sends a response without a content-type

Reply:
actually, that would still perform better, because every `.` is a perf hit using `getattribute` so your `and` here is doubling the number of times `response.headers.get('content-type')` occurs.

Reply:
Well, optimally, by locality of reference, the perf hit would be negligible, but we're dealing with python so performance^TM 🙃